File: FRONTEND CLIENT/HMS_CaseStudy_FrontEnd/src/services/InventoryService.js
// InventoryService.js
import axios from "axios";

const INVENTORY_API_BASE_URL = "http://localhost:8083/api/inventory";

class InventoryService {
  getAllInventory() {
    return axios.get(INVENTORY_API_BASE_URL);
  }

  getInventoryById(id) {
    return axios.get(`${INVENTORY_API_BASE_URL}/${id}`);
  }

  addInventory(inventory) {
    return axios.post(INVENTORY_API_BASE_URL, inventory);
  }

  updateInventory(id, inventory) {
    return axios.put(`${INVENTORY_API_BASE_URL}/${id}`, inventory);
  }

  deleteInventory(id) {
    return axios.delete(`${INVENTORY_API_BASE_URL}/${id}`);
  }
}

export default new InventoryService();

File: BACKEND SERVER/BackEnd/Inventory-service/src/main/java/com/microservice/inventory/service/InventoryService.java
package com.microservice.inventory.service;

import java.util.List;

import com.microservice.inventory.entity.Inventory;
import com.microservie.inventory.exceptions.NotFoundException;

public interface InventoryService {
	
    Inventory createInventory(Inventory inventory);
    
    
    Inventory getInventoryById(String id) throws NotFoundException;
    
    
    Inventory updateInventory(String id, Inventory inventory) throws NotFoundException;
    
    
    void deleteInventory(String id) throws NotFoundException;
    
    List<Inventory> getAllInventory();

}
